src/reducers/pokemonReducer.ts

```diff
@@ -1,82 +1,87 @@
-import { GET_DATA_DONE, GET_DATA_REQUESTED, GET_DATA_FAILED, GET_POKEMON, ADD_TO_FAVORITES, DELETE_FAV_POKEMON } from '../constants';
+import { GET_DATA_DONE, GET_DATA_REQUESTED, GET_DATA_FAILED, GET_POKEMON, DELETE_FAV_POKEMON, ADD_TO_FAVORITES } from '../constants';
 import { handleActions } from 'redux-actions';
 import { Pokemon } from '../model/Pokemon';
+import { combineReducers } from 'redux';
 
-type State = {
-    isLoading: boolean,
-    isError: boolean,
-    pokemonList: Array<Pokemon>,
-    selectedPokemon: Pokemon | null,
-    favoritesPokemons: Array<Pokemon>
-};
+type pokemonList = Array<Pokemon>;
+type isError = boolean;
+type isLoading = boolean;
+type selectedPokemon = Pokemon | null;
+type favoritesPokemons = Array<Pokemon>
 
 const initialState = {
+    pokemonList: [],
     isLoading: false,
     isError: false,
-    pokemonList: [],
     selectedPokemon: null,
     favoritesPokemons: []
+}
+
+const updateFavorites = (favorites, id) => {
+    let newFavorites = favorites.filter(elem => {
+        return (
+            elem.id !== id
+        )
+    });
+    return newFavorites;
+};
+
+const getPokemonList = (state, action) => {
+   return action.payload.results
+};
+
+const setLoadingTrue = (state, action) => {
+   return true;
+};
+
+const setLoadingFalse = (state, action) => {
+   return false;
+};
+
+const setErrorTrue = (state, action) => {
+   return true;
 };
 
- const deletePokemonFav = (favorites, id) => {
-     let newFavorites = favorites.filter(elem => {
-         return (
-             elem.id !== id
-         )
-     });
-     return newFavorites;
- };
-
- const getDataDone = (state, action) => {
-   return {
-     ...state,
-     pokemonList: action.payload.results,
-     isLoading: false
-   };
- };
-
- const getDataRequested = (state, action) => {
-   return {
-     ...state,
-     isLoading: true
-   };
- };
-
- const getDataFailed = (state, action) => {
-   return {
-     ...state,
-     isLoading: false,
-     isError: true
-   };
- };
-
- const getPokemon = (state, action) => {
-   return {
-     ...state,
-     selectedPokemon: action.payload,
-     isLoading: false
-   };
- };
-
- const addToFavorites = (state, action) => {
-   return {
-     ...state,
-     favoritesPokemons: [ ...state.favoritesPokemons, action.payload]
-   };
- };
-
- const deleteFavPokemon = (state, action) => {
-   return {
-     ...state,
-     favoritesPokemons: deletePokemonFav(state.favoritesPokemons, action.payload)
-   };
- };
-
-export const pokemon = handleActions<State, any> ({
-  [GET_DATA_DONE]: getDataDone,
-  [GET_DATA_REQUESTED]: getDataRequested,
-  [GET_DATA_FAILED]: getDataFailed,
-  [GET_POKEMON]: getPokemon,
+const getPokemon = (state, action) => {
+   return action.payload;
+};
+
+const addToFavorites = (state, action) => {
+   return [...state, action.payload ]
```

Comment:
jeśli `state` jest typu `List` to możesz zrobic 
```
return state.push(action.payload)
```

jeśli to jest `List` z immutable wszystkie operacje na tej liscie zwracają nowy obiekt, czyli nie złamiemy zasady reduxa o nie mutowalnym stanie, jeśli nie wiesz co to immutable i List to daj znać, opowiem Ci o tym